stock_location_limit_product/models/stock_location_limit.py

```diff
@@ -1,33 +1,35 @@
 # Copyright (C) 2019 Open Source Integrators
 # Copyright (C) 2019 Serpent Consulting Services Pvt. Ltd.
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
-
-from odoo import fields, models
+from odoo import api, fields, models, _
 from odoo.addons import decimal_precision as dp
+from odoo.exceptions import ValidationError
 
 
 class StockLocationLimit(models.Model):
     _name = 'stock.location.limit'
     _description = 'Stock Location Limit Product'
     _rec_name = 'product_id'
 
+    @api.onchange('product_id')
+    def onchange_uom_id(self):
+        self.uom_id = self.product_id.uom_id.id
+
+    @api.constrains('product_id', 'uom_id')
+    def check_uom_id(self):
```

Comment:
If uom_id is related field then we don't need to check this constrain.

Reply:
@nikitavaghela uom_id is not a related field.

Reply:
@max3903 Do you need the uom to be different of the product one (taht's why I suggested to make it related) ?

Reply:
@rousseldenis Yes, we can have a product using Unit and a limit with that product using a pack of 6 units.

Reply:
:+1: Ok, I see. So, forget about my remark. Your implementation made sense.